engine/src/main/java/io/camunda/zeebe/engine/processing/scheduled/DueDateChecker.java

```diff
@@ -86,23 +82,20 @@ private Duration calculateDelayForNextRun(final long dueDate) {
 
   @Override
   public void onRecovered(final ReadonlyStreamProcessorContext processingContext) {
-    actor = processingContext.getActor();
+    scheduleService = processingContext.getScheduleService();
     streamWriter = processingContext.getLogStreamWriter();
     // check if timers are due after restart
     triggerEntities();
   }
 
   @Override
   public void onPaused() {
-    if (scheduledTimer != null) {
-      scheduledTimer.cancel();
-      scheduledTimer = null;
-    }
+    checkerRunning = false;
```

Comment:
🔧 I feel like `checkerRunning` is doing 2 things. It might be better to split these concerns:
 - if the checker still has a next check scheduled, then it might need to schedule one even further in the future depending on the due date (line 51)
 - if the actor is paused, the due date checker should stop rescheduling.